runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java

```diff
@@ -401,4 +426,51 @@ private static VarPlayer endGoalVarpForSkill(final Skill skill)
 				return null;
 		}
 	}
+
+	@Schedule(
+		period = 1,
+		unit = ChronoUnit.SECONDS
+	)
+	public void tickSkillTimes()
+	{
+		final Instant now = Instant.now();
+		xpPauseState.tick(now);
+
+		if (lastTick == null)
+		{
+			lastTick = now;
+			return;
+		}
+
+		final Duration tickDuration = Duration.between(lastTick, now);
```

Comment:
I dont really like this ticking and sending diffs between ticks here. Cant you just send actual time and handle it properly?

Reply:
Ultimately all calculations are based on a `Duration`.

If I want to pause things between skills independently, then I can't pass time, I have to pass durations.

Reply:
An alternate way would be to pass milliseconds, so `Duration` is out of the picture entirely except for string formatting.

Reply:
By bringing this up, I noticed a bug where I forgot to update the lastTick to now.
That would have been embarrassing.